flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java

```diff
@@ -1294,7 +1294,7 @@ private void removeLocalhostBindHostSetting(
     private void setTokensFor(ContainerLaunchContext containerLaunchContext) throws Exception {
         LOG.info("Adding delegation tokens to the AM container.");
 
-        Credentials credentials = UserGroupInformation.getCurrentUser().getCredentials();
```

Comment:
I've modified the code not to change the current user's credentials (since the framework is partially done).